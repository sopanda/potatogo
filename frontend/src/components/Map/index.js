export { Map as default } from './Map';
