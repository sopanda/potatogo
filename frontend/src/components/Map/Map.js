import React, { forwardRef, memo } from 'react';
import { Map as LeafletMap, TileLayer } from 'react-leaflet';
import { Markers } from 'components/Markers';
import styled from 'styled-components';
import { MAPS_URL } from 'constants/maps';

const MapWrapper = styled(LeafletMap)`
  position: relative;
  width: 100%;
  min-height: 500px;

  & .leaflet-control-container,
  & .leaflet-routing-container-hide {
    display: none !important;
  }
`;

export const Map = memo(
  forwardRef((props, ref) => {
    const { latitude, longitude, children, type, addMarker, markers } = props;
    return (
      <MapWrapper
        center={[latitude, longitude]}
        zoom={15}
        attributionControl
        zoomControl
        doubleClickZoom
        scrollWheelZoom
        dragging
        animate
        easeLinearity={0.35}
        onClick={e => type === 'create' && addMarker(e)}
        ref={ref}
      >
        <TileLayer url={MAPS_URL.WIKIMEDIA} />
        {markers && type === 'home' && <Markers items={markers} />}
        {type === 'create' && children}
      </MapWrapper>
    );
  })
);
