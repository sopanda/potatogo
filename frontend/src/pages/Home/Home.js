import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Map from 'components/Map';
import { Loader } from 'components/Loader';
import { getAllOffers } from 'store/Offer/actions';

const Home = () => {
  const { longitude, latitude } = useSelector(state => state.location);
  const dispatch = useDispatch();
  const { data } = useSelector(state => state.offer);
  useEffect(() => {
    dispatch(getAllOffers());
  }, [dispatch]);

  return longitude && latitude ? (
    <Map
      longitude={longitude}
      latitude={latitude}
      data-test='Map'
      type='home'
      markers={data}
    />
  ) : (
    <Loader />
  );
};

export default Home;
