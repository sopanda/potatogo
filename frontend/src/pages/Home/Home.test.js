import React from 'react';
import { shallow } from 'enzyme';
import Home from './';
import Map from 'components/Map';

describe('Navigation component tests', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Home />);
  });

  it('should render correctly with no props', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render a map', () => {
    expect(component.find(Map)).toHaveLength(1);
  });
});
