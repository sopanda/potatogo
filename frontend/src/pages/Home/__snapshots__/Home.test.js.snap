// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component tests should render correctly with no props 1`] = `
<Fragment>
  <Memo()
    coords={
      Object {
        "latitude": 50.041187,
        "longitude": 21.999121,
      }
    }
    data-test="Map"
  />
  <r
    enableHighAccuracy={false}
    lazy={false}
    maximumAge={0}
    onError={[Function]}
    onSuccess={[Function]}
    timeout={Infinity}
  />
</Fragment>
`;
